### DISCUSSION TITLE
Share relay if subscriber is offline or slow

### DISCUSSION THREAD

**Discussion by jon4hz** [coinfoundry/miningcore#1437] [Q&A] [answered]
Hi,
I have a question regarding the zeromq implementation for the share relay. How exactly does the share relay (publisher) behave if the subscriber is offline or can't process the messages fast enough? Does the publisher build up a queue locally or are these shares going to the void?

**Comment by oliverw** [comment, marked as answer]
https://stackoverflow.com/a/16432160/88513

MC does not configure HWM so my assumption that offline subscribers are going to miss messages published in the meantime and the publisher does not queue up outbound messages.

**Reply by jon4hz** [reply]
okay, thanks for the answer 👍

**Reply by oliverw** [reply]
Fortunately I never ran into any such problems in years of running pools. ZMQ has been rock-solid.

**Reply by jon4hz** [reply]
The link you added mentions that ZMQ 3.0+ implements a default HWM. Which zmq version does miningcore use?

**Reply by oliverw** [reply]
Package `libzmq5` on Linux. Some 4.x version on Windows for development.

**Reply by jon4hz** [reply]
Isn't v4.3.4 latest? According to the githib releases.

**Reply by oliverw** [reply]
https://packages.debian.org/bullseye/libzmq5
